proof/courier.go

```diff
@@ -1012,20 +1106,27 @@ func (c *UniverseRpcCourier) DeliverProof(ctx context.Context,
 			ScriptKey: *proofAsset.ScriptKey.PubKey,
 			OutPoint:  &outPoint,
 		}
-		err = c.deliveryLog.StoreProofDeliveryAttempt(ctx, loc)
-		if err != nil {
-			return fmt.Errorf("unable to log proof delivery "+
-				"attempt: %w", err)
-		}
 
-		// Submit proof to courier.
-		_, err = c.client.InsertProof(ctx, &unirpc.AssetProof{
-			Key:       &universeKey,
-			AssetLeaf: &assetLeaf,
-		})
+		// Setup delivery routine and start backoff procedure.
+		deliverFunc := func() error {
+			// Submit proof to courier.
+			_, err = c.client.InsertProof(ctx, &unirpc.AssetProof{
```

Comment:
This only adds back off for proof insertion. What we've seen fail is that the receiver attempts to look up, but maybe got the block before the sender, then never retries again.

We also need the receiver to properly retry.